### PR TITLE
[AIRFLOW-3303] Deprecate old UI in favor of FAB

### PR DESCRIPTION
**Pull request by verdan**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW-3303) issues and references them in the PR title. 

### Description

- [x] We are using two different versions of UI in Apache Airflow. Idea is to deprecate and remove the older version of UI and use the new Flask App Builder (RBAC) version as the default UI from now on. (most probably in release 2.0.x)
This PR removes the old UI and renames the references of `www_rbac` to `www`. 

### Tests

- [x] Skipped some of the test case classes as these were purely using the older version of application and configurations. 

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality

- [x] Passes `flake8`

**All the help with manual testing would be highly appreciated 🙂**

### CODE REVIEW

**Comment by bolkedebruin** [comment]
@ashb @Fokko what do you say? Let's have this?

File: UPDATING.md
Comment (ashb) [inline UPDATING.md]: Probably worth calling out explicitly that custom auth backends will need re-writing to target FAB

**Review by ashb — DISMISSED** [review]
I'm in favour of this change, but I think there is a bit more work needed around the validation/config

For instance this block https://github.com/apache/incubator-airflow/blob/master/airflow/configuration.py#L173-L188 doesn't make sense anymore, and it possibly makes sense to warn if the old config settings are found.

File: airflow/contrib/plugins/metastore_browser/main.py
Comment (ashb) [inline airflow/contrib/plugins/metastore_browser/main.py:89]: This change seems unrelated - we're still importing wwwutils...?

**Comment by ashb** [comment]
The metastore plugin won't work (be visible) anymore either - it still has `flask_blueprints` and it turns out that we don't integrate those blueprints in the RBAC.

File: airflow/contrib/plugins/metastore_browser/main.py
Comment (verdan) [reply airflow/contrib/plugins/metastore_browser/main.py:89]: in the rbac version the `gzipped` decorator is not in the utils.py, instead it was moved to the decorators.py

Comment (ashb) [reply airflow/contrib/plugins/metastore_browser/main.py:89]: Ah!

**Comment by Fokko** [comment]
@verdan Can you rebase? Would be good to get this merged, since we're now doing duplicate work :-)

File: UPDATING.md
Comment (verdan) [reply UPDATING.md]: good point, updated.

**Comment by verdan** [comment]
@ashb, to add the warnings for old config and integrate metastore blueprints for RBAC, I'd suggest to create new JIRA tickets, as both of these seems like new features. thoughts ?

**Comment by verdan** [comment]
@Fokko @bolkedebruin ready for review.

**Comment by Fokko** [comment]
One more rebase?

**Comment by verdan** [comment]
@Fokko two more rebase done 😉

File: airflow/bin/cli.py
Comment (Fokko) [inline airflow/bin/cli.py]: Why is this commented out?

**Review by Fokko — COMMENTED** [review]
One small comment @verdan

File: airflow/bin/cli.py
Comment (verdan) [reply airflow/bin/cli.py]: Ah, I need to remove this line. As these settings were specific to the older version, I was testing things by commenting out code. Will remove this once the review is complete.

Comment (Fokko) [reply airflow/bin/cli.py]: Please do

**Comment by Fokko** [comment]
@ashb Are you okay with merging this on a short term? Otherwise, we'll end up with merge conflicts again.

File: airflow/bin/cli.py
Comment (verdan) [reply airflow/bin/cli.py]: Done!!

**Comment by kaxil** [comment]
Sorry @verdan , this has been a pain for you to ask rebasing again and again but can you rebase it again? 

A note to all committers - let's get this PR in first before merging anymore PRs. @feng-tao @Fokko @ashb Do you agree guys?

**Comment by feng-tao** [comment]
@kaxil sorry, just saw this. I am ok with your proposal. But based on @ashb  last comment, it seems that we still have some work to do for this pr besides rebase. Does it get resolved? Do we have a time estimate on when it could be finished? If it is for a short period of time, we could go ahead with your approach :)

**Comment by kaxil** [comment]
@feng-tao Agreed, let's wait for @ashb to review first. And once his review is done and @verdan rebases we can then stop merges. Let's continue merging till then :)

**Comment by feng-tao** [comment]
@kaxil , do you want to include this pr in the 10.2 release? I felt this pr seems to fit for a major release better(e.g 11.0). And I think we should send out an email to dev email alias and see if there are any concern on the community(e.g if there are any UI issues, feature disparity bugs for new UI etc.). What do you think?

**Comment by verdan** [comment]
@feng-tao I already started a thread on dev email about this change the time I opened this PR. 
The title of that email was *Deprecate legacy UI in favor of FAB RBAC*

@kaxil yes sure, I'll do the rebase when we all agree to merge this change. (with any possible following changes)

**Comment by feng-tao** [comment]
thanks @verdan for the hard work. And sorry for missing that email previously. I think as we discuss here, IMO we are ok to remove it from master branch once the comment is addressed (whether to pull it in 10.2 or not is TBD).

**Comment by kaxil** [comment]
@feng-tao No, this has to be in a major release (probably 2.0)

**Comment by bolkedebruin** [comment]
Sorry @verdan can you rebase again? I will merge ASAP

**Comment by ashb** [comment]
My comment about the plugin is still un-resolved - the metastore plugin (at least) won't work with the Flask-AppBuilder based UI I think

**Comment by bolkedebruin** [comment]
@ashb I agree, but that is not the issue of this PR, although we should preferably fix it before a release. Why not have a blocker JIRA for 2.0 that raises this issue?

**Comment by ashb** [comment]
Created https://issues.apache.org/jira/browse/AIRFLOW-3684

**Comment by kaxil** [comment]
Can we merge this one before any other PR?

@ashb @feng-tao @Fokko @ashb

**Comment by kaxil** [comment]
Thanks, @verdan and sorry for having to rebase & fix conflicts for the nth time.

Appreciate it.

**Comment by Fokko** [comment]
Thanks @verdan

**Comment by verdan** [comment]
relieved

**Comment by echo-ray** [comment]
the new UI with role manager is pretty nice, but I have took lot of time to get it there.
Hope the new UI would be default in the near future : )

**Comment by diederikwp** [comment]
I have a problem where the webserver keeps repeating "some workers are starting up, waiting...", and I think it might be because of these changes.

It seems that the worker processes no longer get the proper prefix after initialization, because the file gunicorn_config.py was deleted from airflow/www. Before deletion, it contained the post_worker_init function: https://github.com/apache/airflow/blob/cb8b2a1dc64c3ea6ba445893c65c6c953dfb476a/airflow/www/gunicorn_config.py#L25-L28 and it was part of run_args in def webserver in cli.py (and therefore also used by the FAB UI):
https://github.com/apache/airflow/blob/cb8b2a1dc64c3ea6ba445893c65c6c953dfb476a/airflow/bin/cli.py#L874-L883

After your changes, get_num_ready_workers still expects the prefix: https://github.com/apache/airflow/blob/c030729dcbe35a2a4869e46b57a761c8376ede8d/airflow/bin/cli.py#L691-L704

@verdan Could you please confirm/refute

**Comment by feng-tao** [comment]
And also why we rename www_rbac to www which causes all the git history missing?

**Comment by kaxil** [comment]
@feng-tao I agree it is not ideal. In the meantime, you can use this https://github.com/apache/airflow/commits/08eba5241cdfe7449c8f546715401743a5cf0d64/airflow/www_rbac to view git history.

**Comment by verdan** [comment]
@diederikwp yup, you are right. 👍 
I'll open a new PR with the fix.

**Pull request merged**
